### ISSUE TITLE
Problem with image.php

### ISSUE THREAD

**Issue by renaudkaczmarek** [contao/core#7512]
CONTAO 3.4.0
Images generation doesn't trigger when directory doesn't exist in assets.
(I use vagrant on windows.)

Here is my modification to fix it 

\system\modules\core\library\Contao\Image.php

L436

```php
// Custom target (thanks to Tristan Lins) (see #4166)
if ($this->getTargetPath() && !$this->getForceOverride())
{
    if (file_exists(TL_ROOT . '/' . $this->getTargetPath()) && $this->fileObj->mtime <= filemtime(TL_ROOT . '/' . $this->getTargetPath()))
        {
            $this->resizedPath = \System::urlEncode($this->getOriginalPath());
            return $this;
        }
}
//ADD THIS==============================
else{				
    $dir = dirname(TL_ROOT . '/' . $this->getCacheName());
    if (!is_dir($dir)) {
        if (false === @mkdir($dir, 0777, true)) {
            \System::log(sprintf('Could not create directory "%s".', $dir), 'Controller getImage()', TL_ERROR);
        }
    }
}
//END ADD THIS==========================
```

I hope it's good and it will help :)

I had the same problem on previous Contao version (< 3.4 )

\system\modules\core\library\Contao\Image.php

L140 

```php
// Backwards compatibility
		if ($mode == 'crop')
		{
			$mode = 'center_center';
		}

		$strCacheKey = substr(md5('-w' . $width . '-h' . $height . '-' . $image . '-' . $mode . '-' . $objFile->mtime), 0, 8);
		$strCacheName = 'assets/images/' . substr($strCacheKey, -1) . '/' . $objFile->filename . '-' . $strCacheKey . '.' . $objFile->extension;

//ADD THIS==============================
		$dir = dirname(TL_ROOT . '/' . $strCacheName);
        if (!is_dir($dir)) {
            if (false === @mkdir($dir, 0777, true)) {
                \System::log(sprintf('Could not create directory "%s".', $dir), 'Controller getImage()', TL_ERROR);
            }
        }
//END ADD THIS==========================
```

**Comment by renaudkaczmarek** [comment]
Hello Leo, 
maybe it's because we have some process when we create a project from an existing one. We don't start every new project from the core distribution.
Could this explain my need to this modification ?

I'm sorry i'm not an expert, maybe it's not the best solution, but I just try to make things work. :(

**Comment by leofeyer** [comment]
This has been fixed in Contao 3.5.5.

**Comment by renaudkaczmarek** [comment]
Nice!
